### PR TITLE
Bug 1968552: Wait for ISO for 1 minute before using it

### PR DESCRIPTION
**Pull request by mkowalski**
# Description

Because in some cases we do not have a control over the order of
creation between different resources, it helps to wait some time before
BMAC is using the generated ISO image so that we are sure all the
configuration is applied.

Closes: [OCPBUGSM-30125](https://issues.redhat.com/browse/OCPBUGSM-30125)

# What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

# How was this code tested?

- [ ] assisted-test-infra environment
- [x] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Manual test

```
  conditions:
    - lastTransitionTime: '2021-06-22T12:38:46Z'
      message: Image has been created
      reason: ImageCreated
      status: 'True'
      type: ImageCreated
  createdTime: '2021-06-22T12:38:46Z'
  isoDownloadURL: >-
    https://assisted-service-assisted-installer-manual-bundle.apps.ostest.test.metalkube.org/api/assisted-install/v1/clusters/71deabca-da7e-4440-bfab-7eb0f7fb4219/downloads/image?api_key=ey[...]VA
```

```
[...]
time="2021-06-22T12:39:07Z" level=info msg="prepare image for cluster 71deabca-da7e-4440-bfab-7eb0f7fb4219" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GenerateClusterISOInternal" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:865" go-id=797 pkg=Inventory request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="Updating timestamp of file /data/discovery-image-71deabca-da7e-4440-bfab-7eb0f7fb4219.iso" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*FSClient).UpdateObjectTimestamp" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/filesystem.go:242" go-id=797 request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="Re-used existing cluster <71deabca-da7e-4440-bfab-7eb0f7fb4219> image" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GenerateClusterISOInternal" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:997" go-id=797 pkg=Inventory request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="ClusterDeployment Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:113" cluster_deployment=single-node cluster_deployment_namespace=assisted-installer-manual-bundle go-id=818 request_id=9174f497-8df3-4b31-bf0b-13a060f0ff18
time="2021-06-22T12:39:07Z" level=info msg="InfraEnv Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:78" go-id=797 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="ClusterDeployment Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:110" agent_cluster_install=bmac-test agent_cluster_install_namespace=assisted-installer-manual-bundle cluster_deployment=single-node cluster_deployment_namespace=assisted-installer-manual-bundle go-id=818 request_id=9174f497-8df3-4b31-bf0b-13a060f0ff18
time="2021-06-22T12:39:15Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:164" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=06643cbb-728f-4173-a5e5-2cb0c53d6cc6
time="2021-06-22T12:39:15Z" level=info msg="Requeuing reconcileBMH: BMH has InfraEnv with too recent ISO, will requeue with delay" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:584" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=06643cbb-728f-4173-a5e5-2cb0c53d6cc6
time="2021-06-22T12:39:15Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:161" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=06643cbb-728f-4173-a5e5-2cb0c53d6cc6
[...]
time="2021-06-22T12:39:46Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:164" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=5be4cf59-3d71-4176-a74a-d1f36c2a8e80
time="2021-06-22T12:39:46Z" level=info msg="Image URL has been set in the BareMetalHost  assisted-installer-manual-bundle/ostest-extraworker-0" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:611"
time="2021-06-22T12:39:46Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:161" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=5be4cf59-3d71-4176-a74a-d1f36c2a8e80
[...]
time="2021-06-22T12:40:09Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:164" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=41b36675-faa7-40e5-9d97-dcd4a4178894
time="2021-06-22T12:40:09Z" level=info msg="Stopping reconcileBMH: BMH and InfraEnv ISO are in sync" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:587" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=41b36675-faa7-40e5-9d97-dcd4a4178894
time="2021-06-22T12:40:09Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:161" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=41b36675-faa7-40e5-9d97-dcd4a4178894
[...]
```

# Assignees

Please, add one or two reviewers that could help review this PR. Use `/assign` if you want to assign
this PR directly to someone.

/cc @flaper87 
/cc @mhrivnak 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

### CODE REVIEW

**Comment by mkowalski** [comment]
/hold

Waiting for https://github.com/openshift/assisted-service/pull/1975 so that this can be rebased on top

**Comment by mkowalski** [comment]
/cherry-pick ocm-2.3

Backport BZ - https://bugzilla.redhat.com/show_bug.cgi?id=1969367

**Comment by mkowalski** [comment]
/retitle Bug 1968552: Wait for ISO for 1 minute before using it

**Comment by flaper87** [comment]
This looks good to me. Do we want to rebase this? I believe the parent PR landed.

**Comment by mkowalski** [comment]
Nope, https://github.com/openshift/assisted-service/pull/1975 is still not merged

File: internal/controller/api/v1beta1/infraenv_types.go
Comment (mhrivnak) [inline internal/controller/api/v1beta1/infraenv_types.go:113]: suggest `CreatedTime` based on the [api convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions): "The name of a field that specifies the time at which something occurs should be called `somethingTime`."

**Review by mhrivnak — CHANGES_REQUESTED** [review]
If I understand correctly, this will requeue every 20 seconds and check if the 60 seconds have elapsed. Is that right? Can we just do a single `requeueAfter` for the actual amount of time that needs to elapse for the 60 seconds to be done?

File: internal/controller/api/v1beta1/infraenv_types.go
Comment (mhrivnak) [reply internal/controller/api/v1beta1/infraenv_types.go:113]: nevermind, I'll take this to that PR which I see now that you are waiting on.

Comment (mkowalski) [reply internal/controller/api/v1beta1/infraenv_types.go:113]: Correct, I just have this commit so that my PR even compiles, but it’ll disappear once I rebase

**Review by nmagnezi — CHANGES_REQUESTED** [review]
https://github.com/openshift/assisted-service/pull/1975 just got merged.
Note that per https://github.com/openshift/assisted-service/pull/1975#discussion_r654589947, I changed `CreatedAt` to `CreatedTime` so you need to change it here as well.

**Comment by flaper87** [comment]
> If I understand correctly, this will requeue every 20 seconds and check if the 60 seconds have elapsed. Is that right? Can we just do a single `requeueAfter` for the actual amount of time that needs to elapse for the 60 seconds to be done?

+1 

Since this function is called in the InfraEnv events watcher, I think requeuingAfter `InfraEnvImageCooldownPeriod` is probably better as it will reduce the reconcile load

**Comment by mkowalski** [comment]
@mhrivnak @flaper87 this is a valid point, we do not want to run the loop if we know beforehand it's going to be terminated anyway. I have just modified the logic so that we requeue for the time needed for the image to be ready and start reconciling only afterwards.

**Review by mhrivnak — APPROVED** [review]

**Comment by mkowalski** [comment]
/hold

I want to do one more manual test run before we merge that

**Comment by mkowalski** [comment]
What I have noticed in a manual run...

* create InfraEnv
* when ISO is still not 60s old, create BareMetalHost
* the node gets powered on without any ISO mounted and the following appears in the log
```
time="2021-06-21T17:13:21Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:145" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=696 request_id=4189d963-8c45-4440-8ae1-eed3fab9420f
time="2021-06-21T17:13:21Z" level=info msg="Stopping reconcileBMH: Either the InfraEnv image is not ready, the image is not at least 60 seconds old, or there is nothing to update." func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:564" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=696 request_id=4189d963-8c45-4440-8ae1-eed3fab9420f
time="2021-06-21T17:13:21Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:142" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=696 request_id=4189d963-8c45-4440-8ae1-eed3fab9420f
```
* after the image is old enough, BMH is never reconciled
* trigger InfraEnv regenerate new image, the following appears
```
time="2021-06-21T17:20:13Z" level=info msg="Image myinfraenv being prepared for cluster " func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).handleEnsureISOErrors" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:349" go-id=748 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=aa82423e-e67d-4882-97ff-fd98e4f7f743
time="2021-06-21T17:20:13Z" level=info msg="InfraEnv Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:78" go-id=748 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=aa82423e-e67d-4882-97ff-fd98e4f7f743
time="2021-06-21T17:20:33Z" level=info msg="InfraEnv Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:81" go-id=748 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=236118fa-5291-4762-9ee8-e0c419309770
time="2021-06-21T17:20:33Z" level=info msg="the amount of nmStateConfigs included in the image is: 1" func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).ensureISO" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:288" go-id=748 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=236118fa-5291-4762-9ee8-e0c419309770
time="2021-06-21T17:20:33Z" level=info msg="prepare image for cluster c5c2f47f-1145-41c7-ae78-6a4871b2ea58" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GenerateClusterISOInternal" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:865" go-id=748 pkg=Inventory request_id=236118fa-5291-4762-9ee8-e0c419309770
time="2021-06-21T17:20:33Z" level=info msg="Updating timestamp of file /data/discovery-image-c5c2f47f-1145-41c7-ae78-6a4871b2ea58.iso" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*FSClient).UpdateObjectTimestamp" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/filesystem.go:242" go-id=748 request_id=236118fa-5291-4762-9ee8-e0c419309770
time="2021-06-21T17:20:33Z" level=info msg="Re-used existing cluster <c5c2f47f-1145-41c7-ae78-6a4871b2ea58> image" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GenerateClusterISOInternal" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:997" go-id=748 pkg=Inventory request_id=236118fa-5291-4762-9ee8-e0c419309770
time="2021-06-21T17:20:33Z" level=info msg="ClusterDeployment Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:113" cluster_deployment=single-node cluster_deployment_namespace=assisted-installer-manual-bundle go-id=719 request_id=c3c4ef45-bd2e-44f6-9ae6-7d8e0bb6b6fa
time="2021-06-21T17:20:33Z" level=info msg="InfraEnv Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:78" go-id=748 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=236118fa-5291-4762-9ee8-e0c419309770
time="2021-06-21T17:20:33Z" level=info msg="ClusterDeployment Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:110" agent_cluster_install=bmac-test agent_cluster_install_namespace=assisted-installer-manual-bundle cluster_deployment=single-node cluster_deployment_namespace=assisted-installer-manual-bundle go-id=719 request_id=c3c4ef45-bd2e-44f6-9ae6-7d8e0bb6b6fa
```
* BMH is never reconciled again, the node is all the time up without any ISO mounted

So it seems for an unknown yet reason the BMH reconciliation is never called again...

**Comment by mkowalski** [comment]
/lgtm cancel

I have noticed the `reconcileResult` implementation in the current form is not supporting requeuing, instead it transparently drops the `time.Duration` field. I'll implement a `reconcileRequeue` for this scenario and push a new version.

**Comment by mkowalski** [comment]
With the introduced change we are now requeuing and the image is set for the BMH only after the cooldown period passes. Log below:

```
  conditions:
    - lastTransitionTime: '2021-06-22T12:38:46Z'
      message: Image has been created
      reason: ImageCreated
      status: 'True'
      type: ImageCreated
  createdTime: '2021-06-22T12:38:46Z'
  isoDownloadURL: >-
    https://assisted-service-assisted-installer-manual-bundle.apps.ostest.test.metalkube.org/api/assisted-install/v1/clusters/71deabca-da7e-4440-bfab-7eb0f7fb4219/downloads/image?api_key=ey[...]VA
```

```
[...]
time="2021-06-22T12:39:07Z" level=info msg="prepare image for cluster 71deabca-da7e-4440-bfab-7eb0f7fb4219" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GenerateClusterISOInternal" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:865" go-id=797 pkg=Inventory request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="Updating timestamp of file /data/discovery-image-71deabca-da7e-4440-bfab-7eb0f7fb4219.iso" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*FSClient).UpdateObjectTimestamp" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/filesystem.go:242" go-id=797 request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="Re-used existing cluster <71deabca-da7e-4440-bfab-7eb0f7fb4219> image" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).GenerateClusterISOInternal" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:997" go-id=797 pkg=Inventory request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="ClusterDeployment Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:113" cluster_deployment=single-node cluster_deployment_namespace=assisted-installer-manual-bundle go-id=818 request_id=9174f497-8df3-4b31-bf0b-13a060f0ff18
time="2021-06-22T12:39:07Z" level=info msg="InfraEnv Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*InfraEnvReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/infraenv_controller.go:78" go-id=797 infra_env=myinfraenv infra_env_namespace=assisted-installer-manual-bundle request_id=2301e879-d4ea-4608-8c9f-949f5c2703c3
time="2021-06-22T12:39:07Z" level=info msg="ClusterDeployment Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:110" agent_cluster_install=bmac-test agent_cluster_install_namespace=assisted-installer-manual-bundle cluster_deployment=single-node cluster_deployment_namespace=assisted-installer-manual-bundle go-id=818 request_id=9174f497-8df3-4b31-bf0b-13a060f0ff18
time="2021-06-22T12:39:15Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:164" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=06643cbb-728f-4173-a5e5-2cb0c53d6cc6
time="2021-06-22T12:39:15Z" level=info msg="Requeuing reconcileBMH: BMH has InfraEnv with too recent ISO, will requeue with delay" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:584" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=06643cbb-728f-4173-a5e5-2cb0c53d6cc6
time="2021-06-22T12:39:15Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:161" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=06643cbb-728f-4173-a5e5-2cb0c53d6cc6
[...]
time="2021-06-22T12:39:46Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:164" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=5be4cf59-3d71-4176-a74a-d1f36c2a8e80
time="2021-06-22T12:39:46Z" level=info msg="Image URL has been set in the BareMetalHost  assisted-installer-manual-bundle/ostest-extraworker-0" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:611"
time="2021-06-22T12:39:46Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:161" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=5be4cf59-3d71-4176-a74a-d1f36c2a8e80
[...]
time="2021-06-22T12:40:09Z" level=info msg="BareMetalHost Reconcile started" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:164" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=41b36675-faa7-40e5-9d97-dcd4a4178894
time="2021-06-22T12:40:09Z" level=info msg="Stopping reconcileBMH: BMH and InfraEnv ISO are in sync" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).reconcileBMH" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:587" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=41b36675-faa7-40e5-9d97-dcd4a4178894
time="2021-06-22T12:40:09Z" level=info msg="BareMetalHost Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMACReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/bmh_agent_controller.go:161" bare_metal_host=ostest-extraworker-0 bare_metal_host_namespace=assisted-installer-manual-bundle go-id=786 request_id=41b36675-faa7-40e5-9d97-dcd4a4178894
[...]
```

File: internal/controller/controllers/bmh_agent_controller_test.go
Comment (flaper87) [inline internal/controller/controllers/bmh_agent_controller_test.go:347]:
```suggestion
			It("should not reconcile BMH if the updated image has not been around longer than the grace period", func() {
```

Or some shorter version of that :D 

We can make it explicit that there is a grace period that we need wait for

File: internal/controller/controllers/bmh_agent_controller.go
Comment (flaper87) [inline internal/controller/controllers/bmh_agent_controller.go:475]: Starting to think this function should return a `reconcileResult` ... but let's keep that as an improvement for 4.9

**Review by flaper87 — CHANGES_REQUESTED** [review]
just a tiny nit that I think can help future contributors (test analysis).

No need to change the return type for the function at this time, just thinking out loud

File: internal/controller/controllers/bmh_agent_controller.go
Comment (mkowalski) [reply internal/controller/controllers/bmh_agent_controller.go:475]: I agree, it's very ugly that in the caller of this function I need to adjust the type based on the value of one of the returned fields (sic!)

File: internal/controller/controllers/bmh_agent_controller_test.go
Comment (mkowalski) [reply internal/controller/controllers/bmh_agent_controller_test.go]: Done

**Review by flaper87 — APPROVED** [review]

**Comment by osherdp** [comment]
/retest
sorry, forgot to wait for ci-tools image to actually be promoted
let's see how it goes now

**Pull request merged**
